### ISSUE TITLE
Errors shown under wrong cell using magics

### ISSUE THREAD

**Issue by piotrszajer** [ipython/ipyparallel#893]
Hi, I'm facing an issue where Traceback is shown under first cell, where IPyParallel is defined, not under the actual cell in which error was raised. 

Steps to reproduce:
1. Terminal 1
```shell
pip install jupyter ipyparallel
ipcluster nbextension enable
jupyter server extension enable ipyparallel
ipcluster start --n=1
```
2. Terminal 2
```shell
jupyter notebook
```

3. Create new notebook
4. Cell 1: `import this` -> works fine
5. Cell 2:  
```python
import ipyparallel as ipp
rc = ipp.Client()
dview = rc[:]
```
 -> works fine
6. Cell 3:
```python
%%px
import os
print(f"Hello from engine {os.getpid()}")
```
 -> works fine
7. Cell 4:
```python
1/0
``` 
-> Traceback is printed under cell 2

What's more, further execution of cell 4 results in adding Traceback under cell 2, when cell 2 is re-run then Traceback stopped to show.

![Screenshot from 2024-09-10 21-17-48](https://github.com/user-attachments/assets/11d970fa-6cad-4066-9014-81def4db329e)
![Screenshot from 2024-09-10 21-18-00](https://github.com/user-attachments/assets/4f4ac754-7bd3-4449-9b88-bfd3ec580a6c)
![Screenshot from 2024-09-10 21-18-11](https://github.com/user-attachments/assets/ea47b24c-5d47-45e3-a41d-7efe7e9e030e)

[BugReproduce.zip](https://github.com/user-attachments/files/16949971/BugReproduce.zip)

Additional: `%pxresult` shows Traceback in cell were it was executed

Any idea how to overcome it?

**Comment by minrk** [comment]
Thanks for reporting! This is fixed by #905 

If you need a workaround today, you can do the same thing with some private API calls:

```python
import sys
from threading import current_thread


def _detach_thread_output(rc):
    thread_ident = current_thread().ident
    for stream in [sys.stdout, sys.stderr]:
        for name in ("_thread_to_parent", "_thread_to_parent_header"):
            mapping = getattr(stream, name, None)
            if mapping:
                mapping.pop(thread_ident, None)

rc._io_loop.add_callback(_detach_thread_output, rc)
```

**Comment by piotrszajer** [comment]
@minrk Thank you very much :)
